Split/Network/Streaming/Synchronizer.swift

```diff
@@ -185,12 +198,26 @@ class DefaultSynchronizer: Synchronizer {
         }
     }
 
-    func pushImpression(impression: Impression) {
+    func pushImpression(impression: KeyImpression) {
+
+        // This should not happen
+        guard let featureName = impression.featureName else {
+            return
+        }
+
         flushQueue.async {
-            if self.impressionsSyncHelper.pushAndCheckFlush(impression) {
-                self.flusherImpressionsRecorderWorker.flush()
-                self.impressionsSyncHelper.resetAccumulator()
+            var impressionToPush = impression
+            if self.isOptimizedImpressionsMode() {
+                impressionToPush = impression.withPreviousTime(self.impressionsObserver.testAndSet(impression: impression))
```

Comment:
Just in case, recheck Javi, but I think that `testAndSet` must be called in both debug and optimized modes, to set the `pt` of impressions.

Reply:
Yup, fixed doc